caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -178,6 +180,22 @@ def load(cls, model_path: str, *args, **kwargs) -> "EmbeddingModule":
 
         return cls(model)
 
+    @property
+    def public_model_info(cls) -> Dict[str, Any]:
+        """Helper property to return public metadata about a specific Model. This
+        function is separate from `metdata` as that contains the entire ModelConfig
+        which might not want to be shared/exposed.
+
+        Returns:
+            Dict[str, str]: A dictionary of this models's public metadata
+        """
+        return {
+            "metadata": {
+                "max_seq_length": cls.model.max_seq_length,
+                "sentence_embedding_dimension": cls.model.get_sentence_embedding_dimension(),
+            }
+        }
```

Comment:
we can flatten it out and remove extra `metadata` key from here

Reply:
@flaviabeo , `cls` is conventionally used for `classmethods` (like `load`, `train` etc.).. However, since this is an instance method, we should be using `self` here instead of `cls`

Reply:
Hmm ok. I think there is an issue with that too, it is a property method from the base module, so when I use `self` instead of `cls` if prompts that I am changing the args from the base method -- like the error at the linting right now, using `self`, is:
```bash
caikit_nlp/modules/text_embedding/embedding.py:184:4: W0237: Parameter 'cls' has been renamed to 'self' in overriding 'EmbeddingModule.public_model_info' method (arguments-renamed)
```
Any idea how to go over this?

Reply:
And, when changed to `cls` this is the error:
```bash
caikit_nlp/modules/text_embedding/embedding.py:184:4: E0213: Method 'public_model_info' should have "self" as first argument (no-self-argument)
```